engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/ordering/SortVectorNode.java

```diff
@@ -0,0 +1,800 @@
+
+  public static SortVectorNode build() {
+    return SortVectorNodeGen.create();
+  }
+
+  /**
+   * Sorts a vector with elements that have only Default_Comparator, thus, only builtin types.
+   *
+   * @param self Vector that has elements with only Default_Comparator, that are elements with
+   *     builtin types.
+   * @param ascending -1 for descending, 1 for ascending
+   * @param comparators Vector of comparators, with the same length of self. This is gather in the
+   *     Enso code, because doing that in this builtin would be difficult. If {@code onFunc}
+   *     parameter is not {@code Nothing}, comparators are gathered from the result of {@code
+   *     onFunc} projection.
+   * @param compareFunctions Vector of `Comparator.compare` functions gathered from the comparators
+   * @param byFunc If Nothing, then the default `by` function should be used. The default `by`
+   *     function is `Ordering.compare`.
+   * @param onFunc If Nothing, then the default identity function should be used.
+   * @param problemBehavior A long representation of `Problem_Behavior`. Ignore is 0, Report_warning
+   *     is 1, and Report_Error is 2.
+   * @return A new, sorted vector.
+   */
+  public abstract Object execute(
+      State state,
+      @AcceptsError Object self,
+      long ascending,
+      Object comparators,
+      Object compareFunctions,
+      Object byFunc,
+      Object onFunc,
+      long problemBehavior);
+
+  /**
+   * Sorts primitive values, i.e., values with only Default_Comparator. We can optimize this case.
+   * It is important that `byFunc` is Nothing, i.e., has the default value. In that case, we can
+   * hard code the partial ordering for the primitive values. If `byFunc` is a custom user function,
+   * it can redefine the default partial ordering of the primitive values, which requires
+   * topological sort.
+   */
+  @Specialization(
+      guards = {
+        "interop.hasArrayElements(self)",
+        "areAllDefaultComparators(interop, comparators)",
+        "interop.isNull(byFunc)",
+        "interop.isNull(onFunc)"
+      })
+  Object sortPrimitives(
+      State state,
+      Object self,
+      long ascending,
+      Object comparators,
+      Object compareFunctions,
+      Object byFunc,
+      Object onFunc,
+      long problemBehavior,
+      @Cached LessThanNode lessThanNode,
+      @Cached EqualsNode equalsNode,
+      @Cached HostValueToEnsoNode hostValueToEnsoNode,
+      @Cached TypeOfNode typeOfNode,
+      @Cached AnyToTextNode toTextNode,
+      @CachedLibrary(limit = "10") InteropLibrary interop) {
+    EnsoContext ctx = EnsoContext.get(this);
+    Object[] elems;
+    try {
+      long size = interop.getArraySize(self);
+      assert size < Integer.MAX_VALUE;
+      elems = new Object[(int) size];
+      for (int i = 0; i < size; i++) {
+        if (interop.isArrayElementReadable(self, i)) {
+          elems[i] = hostValueToEnsoNode.execute(interop.readArrayElement(self, i));
```

Comment:
Use of `hostValueToEnsoNode` is good in the context of https://github.com/enso-org/enso/pull/6280#discussion_r1167372095

Reply:
Right. I wrapped all the other element reads via `InteropLibrary` with `HostValueToEnsoNode`.